File: expensereport-kotlin/src/main/kotlin/com/nelkinda/training/model/ExpenseType.kt
package com.nelkinda.training.model

enum class ExpenseType(val stringName: String, val expenseLimit: Int) {
    DINNER("Dinner", 5000),
    BREAKFAST("Breakfast", 1000),
    CAR_RENTAL("Car Rental", 0),
    LUNCH("Lunch", 2000)
}

File: expensereport-c/ExpenseType.c
#include "ExpenseType.h"

#include <limits.h>

const struct ExpenseType
    *const DINNER     = &(const struct ExpenseType) { "Dinner",        5000, true },
    *const BREAKFAST  = &(const struct ExpenseType) { "Breakfast",     1000, true },
    *const CAR_RENTAL = &(const struct ExpenseType) { "Car Rental", INT_MAX, false },
    *const LUNCH      = &(const struct ExpenseType) { "Lunch",         2000, true };
